web/portal/platform/src/entities/Administrator/ForeignKeyResolver.tsx

```diff
@@ -15,9 +18,31 @@ const foreignKeyResolver: foreignKeyResolverType = async function ({
     entityService,
   });
 
+  const brandId = store.getState().list.parentRow?.id;
+  const getAction = store.getActions().api.get;
+  const webPortals: Record<number, WebPortalPropertyList<unknown>> = [];
+
+  promises.push(
+    getAction({
+      path: `${entities.WebPortal.path}?_order[id]=ASC`,
+      params: {
+        brand: brandId,
+        urlType: 'brand',
+        _pagination: false,
+      },
+      cancelToken: cancelToken,
+      successCallback: async (response) => {
+        webPortals.push(...response);
+      },
+    })
+  );
   await Promise.all(promises);
 
-  return data;
+  return data.map((row) => {
+    row.impersonationUrl = webPortals[0].url;
+
+    return row;
+  });
```

Comment:
Typescript types in ForeingKeyResolvers are wrong in general (red underline in the IDE). It should be something like this:
```
   const getAction = store.getActions().api.get;
-  const webPortals: Record<number, WebPortalPropertyList<unknown>> = [];
+
+  type WebPortalCollectionType = Array<Partial<WebPortalPropertyList<EntityValue>>>;
+  const webPortals: WebPortalCollectionType = [];
 
   promises.push(
     getAction({
@@ -32,13 +35,15 @@ const foreignKeyResolver: foreignKeyResolverType = async function ({
       },
       cancelToken: cancelToken,
       successCallback: async (response) => {
-        webPortals.push(...response);
+        webPortals.push(...(response as WebPortalCollectionType));
       },
     })
   );
+
   await Promise.all(promises);
 
-  return data.map((row) => {
+  type AdminCollectionType = Array<Partial<AdministratorPropertyList<EntityValue>>>;
+  return (data as AdminCollectionType).map((row) => {
     row.impersonationUrl = webPortals[0].url;
 
     return row;
```

Reply:
Apart from that, Foreign key resolver is supposed to do just PK => string transformations. So, it's quite unexpected to find this code here.

This should probably be handled in the impersonate component itself, but as it would require a service or some code in the store... I'll leave the decision up to you